packages/realm-react/src/RealmProvider.tsx

```diff
@@ -0,0 +1,113 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import React, { useEffect, useState, useRef } from "react";
+import Realm from "realm";
+import { isEqual } from "lodash";
+
+type ProviderProps = Realm.Configuration;
+
+export function createRealmProvider(
+  realmConfig: Realm.Configuration,
+  RealmContext: React.Context<Realm | null>,
+): React.FC<ProviderProps> {
+  return ({ children, ...restProps }) => {
+    const [realm, setRealm] = useState<Realm | null>(null);
+    const [reconfigure, reconfigureRealm] = useState(0);
```

Comment:
```suggestion
    const [configVersion, setConfigVersion] = useState(0);
```

Perhaps this is closer to the semantics of this variable?

Reply:
Like it...it is now changed